eng/containers/UbuntuNetCoreKeyring/Dockerfile

```diff
@@ -1,24 +1,53 @@
-FROM mcr.microsoft.com/dotnet/sdk:5.0-focal AS build
+FROM mcr.microsoft.com/powershell:7.1.4-ubuntu-focal AS build
 
 ENV \
     NO_AT_BRIDGE=1 \
     DOCKER_CONTAINER_NAME="ubuntu_netcore_keyring" \
-    PSModuleAnalysisCachePath=/var/cache/microsoft/powershell/PSModuleAnalysisCache/ModuleAnalysisCache \
-    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-DotnetCoreSDK-Ubuntu-20.04
+     # Unset ASPNETCORE_URLS from aspnet base image
+    ASPNETCORE_URLS= \
+    # Do not generate certificate
+    DOTNET_GENERATE_ASPNET_CERTIFICATE=false \
+    # SDK version
+    DOTNET_SDK_VERSION=5.0.401 \
+    DOTNET_SDK_VERSION2=3.1.413 \
+    # Enable correct mode for dotnet watch (only mode supported in a container)
+    DOTNET_USE_POLLING_FILE_WATCHER=true \
+    # Skip extraction of XML docs - generally not useful within an image/container - helps performance
+    NUGET_XMLDOC_MODE=skip \
+    # PowerShell telemetry for docker image usage
+    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-DotnetSDK-Ubuntu-20.04 \
+    # Setup Dotnet envs
+    DOTNET_ROOT=/usr/share/dotnet  \
+    PATH=$PATH:usr/share/dotnet    
 
 # Install GNOME keyring
 RUN apt-get update \
     && apt-get install -y \
         libsecret-1-dev \
         dbus-x11 \
         gnome-keyring \
-        python
+        python \
+        curl
 
-# Install PowerShell && .net core runtime 2.1
-RUN wget -q https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb \
-    && dpkg -i packages-microsoft-prod.deb \
-    && apt-get update \
-    && apt-get install -y powershell \
-    && pwsh --version \
-    && apt-get install -y dotnet-runtime-2.1 \
-    && dotnet --list-runtimes
+# Below taken from https://hub.docker.com/_/microsoft-dotnet-sdk
+# 5.0.401-focal-amd64, 5.0-focal-amd64, 5.0.401-focal, 5.0-focal	Dockerfile	Ubuntu 20.04	09/14/2021
+
+# Install .NET SDK
+RUN curl -SL --output dotnet.tar.gz https://dotnetcli.azureedge.net/dotnet/Sdk/$DOTNET_SDK_VERSION/dotnet-sdk-$DOTNET_SDK_VERSION-linux-x64.tar.gz \
```

Comment:
Is there any reason we cannot use install instructions like https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2104-?

Reply:
I haven't tried, but this approach gives us very specific control over the versioning.

Reply:
It does but it also makes us more fragile with the installation steps and dependencies. I think you can still specify the versions when using the dotnet instructions as well.

Reply:
The steps are definitely more complex, but it seems to be how most of the reference docker container examples are written. I'm open to trying to revert to the simpler install approach if you feel strongly about it.

Reply:
I don't feel strongly about it my only concern was maintaining these. I was looking at them trying to see if the steps are correct and I wasn't sure and I was also not sure where you got the hashes from so I searched on msdn to see the correct steps for installing dotnet and I found different instructions which is why I started asking questions. Did you find these steps in some other docs somewhere? If so perhaps having a comment with a link to those would also be enough to make sure we maintain these correctly.

Reply:
I put the link in the Docker file https://github.com/Azure/azure-sdk-for-net/pull/24212/files#diff-a65bf4737f710ae4879c5c901b529139118c98f33c08aa018e7870947dcb30e9R32

That site linked me to some reference DockerFiles